plugins/qtcoinrave/qtcoinviewer.cpp

```diff
@@ -640,6 +645,98 @@ void QtCoinViewer::SetUserText(const string& userText)
     _userText = userText;
 }
 
+class SetTextSizeMessage : public QtCoinViewer::EnvMessage
+{
+public:
+    SetTextSizeMessage(QtCoinViewerPtr pviewer, void** ppreturn, double size)
+        : EnvMessage(pviewer, ppreturn, false), _textSize(size) {
+    }
+
+    virtual void viewerexecute() {
+        QtCoinViewerPtr pviewer = _pviewer.lock();
+        if( !pviewer ) {
+            return;
+        }
+        pviewer->_SetTextSize(_textSize);
+        EnvMessage::viewerexecute();
+    }
+
+private:
+    double _textSize;
+};
+
+void QtCoinViewer::SetTextSize(double size)
+{
+    if (_timerSensor->isScheduled() && _bUpdateEnvironment) {
+        EnvMessagePtr pmsg(new SetTextSizeMessage(shared_viewer(), (void**)NULL, size));
+        pmsg->callerexecute(false);
+    }
+}
+
+void QtCoinViewer::_SetTextSize(double size)
+{
+    if ( size > 0 ) {
+        // TODO: use a font that does not rely on hardcoded breakpoints
+        // move down to next text size breakpoint so that message shadow aligns nicely
+        _messagefont->size = _GetTextBaseSize(size);
+        // adjust the text offsets
+        _messageBaseTranslation->translation.setValue(_GetMessageBaseTranslation());
+        _messageShadowTranslation->translation.setValue(_GetMessageShadowTranslation());
+    }
+}
+
+double QtCoinViewer::_GetTextBaseSize(double size)
+{
+    if (size < 14.0) {
+        return 10.0;
+    }
+    if (size < 18.0) {
+        return 14.0;
+    }
+    if (size < 26.0) {
+        return 18.0;
+    }
+    return 26.0;
+}
+
+SbVec3f QtCoinViewer::_GetMessageBaseTranslation()

```

Comment:
i think having some doc covering _GetMessageBaseTranslation and _GetMessageShadowTranslation would be good

Reply:
added more doc on those methods in [9b1f9c6](https://github.com/rdiankov/openrave/pull/1366/commits/9b1f9c6bb1d9ac48235c30039e18bf5634ace930)